lib/service/service.go

```diff
@@ -1150,6 +1150,7 @@ func (process *TeleportProcess) initAuthService() error {
 		CASigningAlg:         cfg.CASignatureAlgorithm,
 		Emitter:              checkingEmitter,
 		Streamer:             events.NewReportingStreamer(checkingStreamer, process.Config.UploadEventsC),
+		PublicAddrs:          cfg.Proxy.PublicAddrs,
```

Comment:
@Joerger This is the proxies public addresses. I think you want the PingResponse to return the public address of the Auth Server right?

Reply:
> @Joerger This is the proxies public addresses. I think you want the PingResponse to return the public address of the Auth Server right?

`PublicAddrs` is supposed to be the Web proxy, I can rename it to `ProxyPublicAddrs` to be more clear. 

> Also keep in mind that this won't work on most deployments. This only works when Auth and Proxy run within the same process. Usually you'll see Auth and Proxy running in different processes.

IIUC, the `Proxy.PublicAddrs` are pulled from the config file on startup. Are you saying that the config files for proxy and auth might be separated? If so, is there any way to retrieve it without major changes to the process initialization?

Reply:
Yes, that field will not be set unless you are running both within the same process.

You're executing in `ServerWithRoles` you should have the `GetProxies` function available to you. Fetch the available proxies and then loop over them and pick the first one where `GetPublicAddr()` returns a valid address.

Just make sure you hit the cache for the `GetProxies` call otherwise you'll be hitting the backend for each call. In `auth.go` you should have `a.GetCache().GetProxies()`, might have to cache if the cache is available within `ServerWithRoles`.